subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/JfrToStacksConverter.java

```diff
@@ -53,14 +54,15 @@
  * Converts JFR recordings to the collapsed stacks format used by the FlameGraph tool.
  */
 class JfrToStacksConverter {
-    public void convertToStacks(IItemCollection recording, File targetFile, Options options) {
-        Map<String, Long> foldedStacks = foldStacks(recording, options);
+    public void convertToStacks(Collection<IItemCollection> recordings, File targetFile, Options options) {
+        Map<String, Long> foldedStacks = foldStacks(recordings, options);
         writeFoldedStacks(foldedStacks, targetFile);
     }
 
-    private Map<String, Long> foldStacks(IItemCollection recording, Options options) {
+    private Map<String, Long> foldStacks(Collection<IItemCollection> recordings, Options options) {
         StackFolder folder = new StackFolder(options);
-        StreamSupport.stream(recording.spliterator(), false)
+        recordings.stream()
+            .flatMap(recording -> StreamSupport.stream(recording.spliterator(), false))
```

Comment:
minor: any reason for not using Guava's `Streams.stream(recording)` here?
(it's not a dependency of the module, though)

Reply:
We haven't upgraded to the Java 8-compatible variant of Guava yet, so that method is not available.